### PR TITLE
feat: improve CLI help commands, add command groups

### PR DESCRIPTION
**Pull request by danielpeach**
## Description

Updates CLI help text, which needed some help. Also groups commands and gives them a better sort.

Before
Screenshot 2023-02-02 at 5 40 04 PM

After
Screenshot 2023-02-02 at 5 39 47 PM

## How has this been tested? How can a reviewer test these changes?

I've manually tested that I can still log in and deploy.

# Have your performed the following tests?

Please confirm you have done the following tests to ensure your pull request is ready to be merged..

- [x] If this change modifies the deployment flow have triggered a deployment using this branch (i.e. `build/dist/darwin_amd64/armory deploy start -f <path to file>`)?
- [x] If this change modifes the login flow have you tried logging in and out with this branch (i.e. `build/dist/darwin_amd64/armory login`) and verified the credentials work?
- [x] Have you verified the specific change made in this PR?

### CODE REVIEW

File: cmd/rootCmd.go [hunk 1]
```diff
@@ -26,23 +26,26 @@ import (
...
+	if err := rootCmd.PersistentFlags().MarkHidden("authToken"); err != nil {
 		return nil
 	}
 
-	clientId := rootCmd.PersistentFlags().StringP("clientId", "c", "", "configure oidc client credentials for Armory CD-as-a-Service API")
-	clientSecret := rootCmd.PersistentFlags().StringP("clientSecret", "s", "", "configure oidc client credentials for Armory CD-as-a-Service API")
-	accessToken := rootCmd.PersistentFlags().StringP("authToken", "a", "", "use an existing access token, rather than client id and secret or user login")
-	verbose := rootCmd.PersistentFlags().BoolP("verbose", "v", false, "show more details")
-	outFormat := rootCmd.PersistentFlags().StringP("output", "o", "text", "Set the output type. Available options: [json, yaml, text].")
+	clientId := rootCmd.PersistentFlags().StringP("clientId", "c", "", "Authenticate using an Armory CD-as-a-Service client ID")
```
Comment (aimeeu) [inline cmd/rootCmd.go:45]:
```suggestion
	clientId := rootCmd.PersistentFlags().StringP("clientId", "c", "", "Authenticate using an Armory CD-as-a-Service Client ID")
```

File: cmd/rootCmd.go [hunk 2]
```diff
@@ -26,23 +26,26 @@ import (
...
+	clientSecret := rootCmd.PersistentFlags().StringP("clientSecret", "s", "", "Authenticate using an Armory CD-as-a-Service client secret")
```
Comment (aimeeu) [inline cmd/rootCmd.go:46]:
```suggestion
	clientSecret := rootCmd.PersistentFlags().StringP("clientSecret", "s", "", "Authenticate using an Armory CD-as-a-Service Client Secret")
```

File: pkg/deploy/deploy.go [hunk 3]
```diff
@@ -146,4 +146,4 @@ type networkError struct {
 
 func (d *networkError) Error() string {
 	return "Unable to reach Armory Continuous Delivery as A Service. Please check your internet connection and try again. "
```
Comment (aimeeu) [inline pkg/deploy/deploy.go:148]:
```suggestion
	return "Unable to reach Armory CD-as-a-Service. Please check your internet connection and try again. "
```

**Review by aimeeu — CHANGES_REQUESTED** [review]
A few minor things. Client ID and Client Secret is how the docs and UI  has those fields

File: cmd/rootCmd.go [hunk 4]
```diff
@@ -58,14 +61,27 @@ func NewCmdRoot(outWriter, errWriter io.Writer) *cobra.Command {
...
 
 	CheckForUpdate(configuration)
+	rootCmd.AddGroup(
+		&cobra.Group{
+			ID:    "deployment",
+			Title: "Deployment Commands:",
+		},
+		&cobra.Group{
+			ID:    "management",
+			Title: "Management Commands:",
```
Comment (th3morg) [inline cmd/rootCmd.go]: Thoughts on calling this "Administrative Commands"?

**Review by th3morg — APPROVED** [review]

File: cmd/rootCmd.go (hunk 4 above)
Comment (danielpeach) [reply cmd/rootCmd.go]: good thinking, I'll call them that

**Review by aimeeu — APPROVED** [review]

**Pull request merged**